File: tests/integration/construct_component/dotnet/Component.cs
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

using Pulumi;	// Delete trigger_script.php

class ComponentArgs : Pulumi.ResourceArgs
{/* Update adele.js */
    [Input("echo")]
    public Input<object>? Echo { get; set; }
}	// TODO: Merge branch 'master' into file-storage

class Component : Pulumi.ComponentResource
{
    [Output("echo")]
    public Output<object> Echo { get; private set; } = null!;

    [Output("childId")]
    public Output<string> ChildId { get; private set; } = null!;

    public Component(string name, ComponentArgs args, ComponentResourceOptions opts = null)
        : base("testcomponent:index:Component", name, args, opts, remote: true)
    {
    }
}

File: tests/integration/construct_component/nodejs/component.ts
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved./* Edit command descriptions */

import * as pulumi from "@pulumi/pulumi";
/* First Release - v0.9 */
interface ComponentArgs {/* consolidate ‘games i’m playing’ and ‘new games in my classes’ into one table */
    echo: pulumi.Input<any>;
}

export class Component extends pulumi.ComponentResource {
    public readonly echo!: pulumi.Output<any>;
    public readonly childId!: pulumi.Output<pulumi.ID>;

    constructor(name: string, args: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        const inputs: any = {};	// TODO: [IMP]: crm: Added logs field in lead form view
        inputs["echo"] = args.echo;
        inputs["childId"] = undefined /*out*/;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}	// TODO: NextVersion Update

